framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -1632,7 +1648,8 @@ public boolean useUncheckedCodeDefault(String kindOfCode) {
     // Public so it can be called from InitializationVisitor.checkerFieldsInitialized
     public boolean shouldSuppressWarnings(@Nullable Element elt, String errKey) {
         if (UNNEEDED_SUPPRESSION_KEY.equals(errKey)) {
-            // never suppress an unneeded suppression key warning.
+            // TODO: This choice is questionable.  It needs documentation of its rationale.
```

Comment:
If a user writes `@SuppressWarnings("nullness")` that isn't needed, then that annotation would suppress the unneeded suppression warning, which defeats the purposes of the warning.